docker/Dockerfile.tests

```diff
@@ -33,3 +29,7 @@ RUN pip install --no-cache-dir --requirement requirements-dev.txt --requirement
 FROM dev as dev-tools
 RUN pip install --no-cache-dir --requirement requirements-dev.txt --requirement reqs/requirements-dev-test.txt \
     --requirement reqs/requirements-dev-tools.txt --constraint constraints-dev.txt
+
+FROM base as build_local
```

Comment:
The stage `FROM build_${SOURCE} AS dev` either reads from `build_local` or `build_github` depending on whether you are building an image with your local environment or from github (controlled by the ARG `SOURCE` passed into the `docker buildx` command). Moving this stage lower will break this.